ssc/cmod_battery_stateful.cpp

```diff
@@ -389,22 +391,25 @@ std::shared_ptr<battery_params> create_battery_params(var_table *vt, double dt_h
     return params;
 }
 
-cm_battery_stateful::cm_battery_stateful():
-        dt_hr(0),
-        control_mode(0){
+cm_battery_stateful::cm_battery_stateful() :
+    dt_hour(0),
+    control_mode(0) {
     add_var_info(vtab_battery_stateful_inputs);
     add_var_info(vtab_battery_state);
 }
 
 cm_battery_stateful::cm_battery_stateful(var_table* vt) :
-        cm_battery_stateful() {
+    cm_battery_stateful() {
     m_vartab = vt;
     try {
         if (!compute_module::verify("precheck input", SSC_INPUT))
             throw exec_error("battery_stateful", log(0)->text);
         dt_hr = as_number("dt_hr");
         control_mode = as_integer("control_mode");
-        params = create_battery_params(m_vartab, dt_hr);
+        params = create_battery_params(m_vartab, dt_hour);
+        if ((params->voltage->dynamic.Vcut > params->voltage->dynamic.Vnom) || (params->voltage->dynamic.Vnom > params->voltage->dynamic.Vexp) || (params->voltage->dynamic.Vexp > params->voltage->dynamic.Vfull)) {
+            throw exec_error("battery_stateful", "Vfull must be greater than Vexp, which must be greater than Vnom, which must be greater than Vcut");
```

Comment:
Can you put these error checks and the other voltage checks from cmod_battery into the voltage_dynamic_t constructor? See `initialize`

Reply:
I added Vcut to the checks in the constructor but there was already one there. Do you mean instead of the checks in cmod_battery and battery_stateful or in addition to?

Reply:
Instead of - if the checks are in voltage_dynamic_t both cmods will run them, rather than having to have the code in two places.